src/battle/action_cmd/power_shock.c

```diff
@@ -2,12 +2,321 @@
 
 #define NAMESPACE action_command_power_shock
 
-INCLUDE_ASM(s32, "battle/action_cmd/power_shock", func_802A9000_42CF10);
+extern s32 D_80108AFC;
+extern s32 D_80108B28;
+extern s32 D_80108B80;
+extern s32 D_80109244;
+extern s32 D_80292974;
+extern s32 D_802943A0;
+extern s16 D_802A9AA2_42D9B2[];
+extern s32 D_802A9B00;
 
-INCLUDE_ASM(s32, "battle/action_cmd/power_shock", func_802A9210_42D120);
+ApiStatus N(CreateHudElements)(Evt* script, s32 isInitialCall) {
+    ActionCommandStatus* actionCommandStatus = &gActionCommandStatus;
+    BattleStatus* battleStatus = &gBattleStatus;
+    HudElement* hudElement;
+    s32 temp_v1;
 
+    battleStatus->unk_82 = 0x64;
+    battleStatus->unk_434 = &D_802943A0;
+
+    if (battleStatus->unk_83 == 0) {
+        battleStatus->actionSuccess = 0;
+        return ApiStatus_DONE2;
+    } else {
+        func_80268858();
+        actionCommandStatus->actionCommandID = ACTION_COMMAND_POWER_SHOCK;
+        actionCommandStatus->state = 0;
+        actionCommandStatus->unk_60 = 0;
+        actionCommandStatus->unk_44 = 0;
+        actionCommandStatus->unk_48 = 0;
+        actionCommandStatus->unk_46 = rand_int(100);
+        actionCommandStatus->unk_6C = 30;
+        actionCommandStatus->hudElementX = -48;
+        actionCommandStatus->unk_68 = 0;
+        actionCommandStatus->unk_66 = 0;
+        D_802A9B00 = 0;
+        actionCommandStatus->hudElementY = 80;
+
+        hudElement = create_hud_element(&D_80108B28);
```

Comment:
Can you type D_80108B28 and all the other hud element scripts as s32 arrays instead and remove the `&`?

Reply:
Done; done for D_802943A0 as well since it looks to be an array of multipliers for how much the bar fills per press indexed by actionCommandStatus->unk_50.